iib/workers/tasks/greenwave.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import json
+import logging
+
+import requests
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import get_image_labels
+
+log = logging.getLogger(__name__)
+
+
+def gate_bundles(bundles, greenwave_payload):
+    """
+    Check if all bundle images have passed gating tests in the CVP pipeline.
+
+    This function queries Greenwave to check if the policies are satisfied for each bundle image.
+
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        be gated.
+    :param dict greenwave_payload: the dict of payload required to query Greenwave to gate bundles.
```

Comment:
I find the argument name and description is misleading since it's missing the `subject_identifier`, so it's not actually the payload.